### PR TITLE
EZP-30024: Simplifies object states table

### PR DESCRIPTION
**Pull request by pawbuj**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-30024](https://jira.ez.no/browse/EZP-30024) [EZP-30055](https://jira.ez.no/browse/EZP-30055)
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Simplifies object states table
ez platform 2019-01-30 10-59-02

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by dew326** [comment]
Can you add some screenshots?

**Comment by pawbuj** [comment]
> Can you add some screenshots?

done

File: src/lib/Form/Type/ObjectState/ContentObjectStateUpdateType.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     return array_filter(
                         $this->objectStateService->loadObjectStates($objectStateGroup),
                         function (ObjectState $objectState) use ($contentInfo, $contentState) {
-                            return $this->permissionResolver->canUser('state', 'assign', $contentInfo, [$objectState]) && $contentState->id !== $objectState->id;
```
Comment (webhdx) [inline src/lib/Form/Type/ObjectState/ContentObjectStateUpdateType.php]: @pawbuj I'm wondering why you had to remove ` && $contentState->id !== $objectState->id`?

Comment (pawbuj) [reply src/lib/Form/Type/ObjectState/ContentObjectStateUpdateType.php]: there is an additional requirement ordered by @SylvainGuittard to display a list of all states

Comment (SylvainGuittard) [reply src/lib/Form/Type/ObjectState/ContentObjectStateUpdateType.php]: That is the main goal of this PR. 
Users were confused to not see all the object states in the dropdown list. So we display all the states that the user has the rights to change

**Pull request merged**